source/BatchedGeometry.cpp

```diff
@@ -212,16 +212,17 @@ uint32 CountUsedVertices(IndexData *id, std::map<uint32, uint32> &ibmap)
 ///
 void BatchedGeometry::extractVertexDataFromShared(const Ogre::MeshPtr &mesh)
 {
-   if (mesh.isNull() || !mesh->sharedVertexData)
-      return;
+	if (!mesh || !mesh->sharedVertexData)
+		return;
 
-   Mesh::SubMeshIterator subMeshIterator = mesh->getSubMeshIterator();
+   Mesh::SubMeshList subMeshList = mesh->getSubMeshes();
 
    // Get shared vertex data
    VertexData *oldVertexData = mesh->sharedVertexData;
 
-   while (subMeshIterator.hasMoreElements()) {
-      SubMesh *subMesh = subMeshIterator.getNext();
+   for (auto subMeshIter = subMeshList.begin(); subMeshIter != subMeshList.end(); ++subMeshIter) {
+	   SubMesh *subMesh = subMeshList.back();

```

Comment:
this is better as you are now actually iterating over the list, but still wrong as you clear it at that.

use: `for(auto subMesh : submeshList)`